runtime/vam/expr/arith.go

```diff
@@ -48,5 +51,90 @@ func (a *Arith) eval(vecs ...vector.Any) vector.Any {
 	if !ok {
 		return vector.NewStringError(a.zctx, coerce.ErrIncompatibleTypes.Error(), lhs.Len())
 	}
-	return f(lhs, rhs)
+	out, ok := f(lhs, rhs)
+	if !ok {
+		return a.evalDivideByZero(kind, lhs, rhs)
+	}
+	return out
```

Comment:
Keep genarithfuncs.go a little simpler by letting the hardware do the check?
```suggestion
	if a.opCode == vector.ArithDiv || a.opCode == vector.ArithMod {
		defer func() {
			if v := recover(); v != nil {
				if err, ok := v.(runtime.Error); ok && err.Error() == "runtime error: integer divide by zero" {
					out = a.evalDivideByZero(kind, lhs, rhs)
				} else {
					panic(v)
				}
			}
		}()
	}
	return f(lhs, rhs)	
```

Reply:
Yeah I was thinking about this sounds good